src/license_manager_cli_ops.py

```diff
@@ -0,0 +1,202 @@
+        self._charm = charm
+
+    def install(self):
+        """Install license-manager-cli and setup ops."""
+
+        # Setup log dir
+        self.setup_log_dir()
+
+        # Setup cache dir
+        self.setup_cache_dir()
+        
+        # Create the virtualenv
+        create_venv_cmd = [
+            self._PYTHON_BIN.as_posix(),
+            "-m",
+            "venv",
+            self._VENV_DIR.as_posix(),
+        ]
+        subprocess.call(create_venv_cmd)
+        logger.debug("license-manager-cli virtualenv created")
+
+        # Ensure pip
+        ensure_pip_cmd = [
+            self._VENV_PYTHON,
+            "-m",
+            "ensurepip",
+        ]
+        subprocess.call(ensure_pip_cmd)
+        logger.debug("pip ensured")
+
+        # Ensure we have the latest pip
+        upgrade_pip_cmd = [
+            self._VENV_PYTHON,
+            "-m",
+            "pip",
+            "install",
+            "--upgrade",
+            "pip",
+        ]
+        subprocess.call(upgrade_pip_cmd)
+
+        # Install license-manager-cli package
+        pip_install_cmd = [
+            self._VENV_PYTHON,
+            "-m",
+            "pip",
+            "install",
+            self._PACKAGE_NAME,
+        ]
+        logger.debug(f"## Running: {pip_install_cmd}")
+        try:
+            out = subprocess.check_output(pip_install_cmd, env={}).decode().strip()
+            logger.debug("license-manager-cli installed")
+            logger.debug(f"## pip install output: {out}")
+        except Exception as e:
+            logger.error(f"Error installing license-manager-cli: {e}")
+            raise Exception("license-manager-cli not installed.")
+
+        # Create the lm-cli script file
+        self.configure_bin_script()
+
+        # Create the etc/default/lm-cli file
+        self.configure_etc_default()
+
+
+    def setup_cache_dir(self):
+        """Set up cache dir."""
+
+        # Delete cache dir if it already exists
+        if self._CACHE_DIR.exists():
+            rmtree(self._CACHE_DIR, ignore_errors=True)
+        # Create a clean cache dir
+        self._CACHE_DIR.mkdir(parents=True)
+        chown(self._CACHE_DIR.as_posix(), self._SLURM_USER, self._SLURM_GROUP)
+        self._CACHE_DIR.chmod(0o700)
+
+
+    def setup_log_dir(self):
+        """Set up log dir."""
+
+        # Create log dir
+        if not self._LOG_DIR.exists():
+            self._LOG_DIR.mkdir(parents=True)
+        chown(self._LOG_DIR.as_posix(), self._SLURM_USER, self._SLURM_GROUP)
+        self._LOG_DIR.chmod(0o700)
+
+
+    def upgrade(self, version: str):
+        """Upgrade license-manager-cli package to specified version."""
+
+        pip_install_cmd = [
+            self._VENV_PYTHON,
+            "-m",
+            "pip",
+            "--upgrade",
+            f"{self._PACKAGE_NAME}=={version}",
+        ]
+
+        out = subprocess.check_output(pip_install_cmd).decode().strip()
+        if "Successfully installed" not in out:
```

Comment:
what do you think about checking the exit code? I think it's better because, since pip is being updated, at some point it is possible to have a release that prints something slightly different but it doesn't mean some error happened

Reply:
> I agree that this is probably the preferred method.
> 
> Howeer, since this is following the pattern of our other charms and we need it in time for the production release, I'm going to create a task to apply this change to all the charms at once.
> 
> Let's leave this as is for now.

the cluster agent follows the pattern I sent. I think it could be changed now

Reply:
> https://app.clickup.com/t/18022949/ARMADA-866

I'll keep it this way for now, and we can change it later when working in this task.